components/HeroBanner/index.vue

```diff
@@ -0,0 +1,130 @@
+          <div class="herobanner__datatable-header-filter-icon">
+            <jds-icon name="filter-outline" size="16px" />
+          </div>
+          <span class="herobanner__datatable-header-filter-text">
+            Filter :
+          </span>
+          <button type="button" class="herobanner__datatable-header-filter-dropdown">
+            Belum ada filter
+          </button>
+        </div>
+        <jds-search
+          v-model="search"
+          small
+          icon
+          :button="false"
+          class="herobanner__datatable-header-search"
+        />
+      </div>
+      <BaseDataTable
+        :headers="headerTableHeroBanner"
+        :items="dataHeroBanner"
+        :loading="$fetchState.pending"
+        :pagination="pagination"
+        @next-page="nextPage"
+        @previous-page="previousPage"
+        @page-change="pageChange"
+        @per-page-change="perPageChange"
+      >
+        <!-- eslint-disable-next-line vue/valid-v-slot -->
+        <template #item.action="{item}">
+          <HeroBannerTableAction :item="item" />
+        </template>
+      </BaseDataTable>
+    </div>
+  </div>
+</template>
+
+<script>
+import { headerTableHeroBanner, headerHeroBanner } from '@/constants/landingPage'
+export default {
+  name: 'ComponentHeroBanner',
+  layout: 'Dashboard',
+  data () {
+    return {
+      search: '',
+      headerTableHeroBanner,
+      headerHeroBanner,
+      data: [],
+      pagination: {
+        currentPage: 1,
+        totalRows: 0,
+        itemsPerPage: 5,
+        itemsPerPageOptions: [],
+        disabled: false
+      },
+      query: {
+        per_page: 5,
+        current_page: 1
+      }
+    }
+  },
+  async fetch () {
+    const response = await this.$axios.get('/pages', { params: this.query })
+    const { data, meta } = response.data
+    this.data = data || []
+    this.pagination.currentPage = meta?.current_page || 1
+    this.pagination.totalRows = meta?.total || 0
+    this.pagination.itemsPerPage = meta?.per_page || this.query.per_page
+  },
+  computed: {
+    dataHeroBanner () {
+      return this.data.map((item) => {
+        return {
+          ...item,
+          status: item.is_active === true ? 'Aktif' : 'Non-Aktif'
+        }
+      })
+    },
+    generateItemsPerPageOptions () {
+      const options = []
+      for (let i = 1; i <= 5; i++) {
+        options.push(this.pagination.itemsPerPage * i)
+      }
+      return options
+    }
+  },
+  mounted () {
+    this.pagination.itemsPerPageOptions = this.generateItemsPerPageOptions
+  },
+  methods: {
+    perPageChange (value) {
+      this.query.per_page = value
+      this.query.current_page = 1
+      this.$fetch()
+    },
+    pageChange (value) {
+      this.query.current_page = value
+      this.$fetch()
+    },
+    nextPage (value) {
+      this.query.current_page = value
+      this.$fetch()
+    },
+    previousPage (value) {
+      this.query.current_page = value
+      this.$fetch()
+    }
```

Comment:
please take notes that every `itemsPerPage` change there will be 2 events that emits from `JdsPagination` so it's more likely will double fetching the API @adzharamrullah

Reply:
I have done refactored it